packages/metro-transform-worker/src/__tests__/index-test.js

```diff
@@ -600,3 +600,18 @@ it('skips minification in Hermes canary transform profile', async () => {
     });"
   `);
 });
+
+it('counts all line endings correctly', async () => {
+  const result = await Transformer.transform(
+    baseConfig,
+    '/root',
+    'local/file.js',
+    'one\rtwo\r\nthree\nfour\u2028five\u2029six',
+    {
+      dev: false,
+      minify: false,
+      type: 'module',
+    },
+  );
+  expect(result.output[0].data.lineCount).toEqual(9);
```

Comment:
I changed the test so rather than comparing to a constant it compares to the result when the input has vanilla \n endings. Should be robust to other transformer changes.